code/components/citizen-server-impl/include/NoticeLogicProcessor.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <StdInc.h>
+#include <ServerInstanceBase.h>
+#include <ResourceManager.h>
+
+#include <json.hpp>
+
+using json = nlohmann::json;
+
+class NoticeLogicProcessor
+{
+private:
+	enum RuleOp
+	{
+		And,
+		Or,
+		Not,
+		NullOrEmpty,
+		Contains,
+		Equals
+	};
+
+	enum RuleType
+	{
+		ConVar,
+		StartedResourceList
+	};
+
+public:
+	NoticeLogicProcessor(fx::ServerInstanceBase* server);
+	bool ProcessNoticeRule(json& ruleRef, uint32_t nestingLevel);
```

Comment:
It looks like this one is not making changes to either the processor instance nor the json file, we could then declare it as: `bool ProcessNoticeRule(const json& ruleRef, uint32_t nestingLevel) const;`.

The static function can then also be changed to a `const json&`  parameter. I don't think `ServerInstanceBase` is correctly set up to be const-qualified yet, otherwise I'd like to see that one `const` as well.

Reply:
Added const where applicable, no part of the current codebase uses ServerInstanceBase in conjunction with const so I left that one as is.